### DISCUSSION TITLE
Triggering Child Tab Field Validation/Errors From React Bootstrap Parent (codesandbox)

### DISCUSSION THREAD

**Discussion by frequencydesign** [react-hook-form/react-hook-form#2340] [Q&A]
Hello and thank you more than I can express.

File/Component Structure:
Grandparent - <Tabs> wrapper component
Parent - Form Container component
Child - Form Elements

Using React Bootstrap Tabs (3.3.7) - when I click between tabs, I need the child form elements to validate/error handle. Currently, Data is saved when switching tabs (back and forth) but errors aren't triggering and and a user can just skip past a tab that has required fields which need to be valid/filled in before passing. Note I'm still (stuck) on react-hook-form": "**^5.7.2**"

Example Codesandbox based off original React-Hook-Form Page Form Wizard example:
https://codesandbox.io/s/react-hook-form-wizard-form-c0j52 

    "bootstrap": "**3.3.7**",
    "react": "**16.8.1**",
    "react-bootstrap": "**^0.32.4**",
    "react-hook-form": "**^5.7.2**"

There is an onSelect ( https://react-bootstrap-v3.netlify.app/components/tabs/#tabs-controlled ) prop in the Grandparent Tabs component which I had hoped I could trigger validation - but it didn't seem to work:
```
  const { handleSubmit, errors, register, watch, triggerValidation } = useForm({});

      <Tabs defaultActiveKey={1} id="contacts-tab" onSelect={triggerValidation}>
```

I think it didn't work because the Errors are in child (grandchild) components but I could easily be wrong.

Love to hear your thoughts - I imagine I'm close.

Thanks everyone.

**Comment by bluebill1049** [comment]
yes, because each tab will get unmounted and inputs will unregister. I think you have two options

1. follow this example: https://codesandbox.io/s/tabs-760h9
2. use v6 shouldUnregister

**Reply by frequencydesign** [reply]
Thank you @bluebill1049 
I looked through all the 6.0.0-rc notes and a few other 6.0.x release notes and thing I'm minority affected by an upgrade.

I tried adding shouldUnregister to the example link you gave, as well as to the example link I gave (upgraded version). Without and without shouldUnregister, I was still able to switch tabs without enforcing required fields. I made a fork. You'll see I added a touch of error messages to make it obvious. Anything I'm missing. Thanks again. https://codesandbox.io/s/tabs-xqmq1?file=/src/Form1.tsx

Also, would I have to manually unregister all 16 or so form elements from each of my tabs if I used shouldUnregister?

**Reply by bluebill1049** [reply]
the example that i posted, doesn't need shouldUnregister, because each tab it's own form, which makes thing easier. you can potentially doing that in your example as well, each tab contains it's own form and validation, and the parent component hosts all the data inputs.

**Comment by Biz-e-Services** [comment]
Hi Bill/RHF Team
I've a slightly different situation. I'm following the React-Hook-Form Page Form Wizard example along with the https://codesandbox.io/s/tabs-760h9?file example, but on my page 2 I'm using Material-UI Tabs (https://material-ui.com/components/tabs/). How would I apply validation to the tabs that are "hidden" I've seen it mentioned that you can't . Can you verify this and are there any workarounds to this? Any suggestions or resources most welcome.

**Reply by bluebill1049** [reply]
have you seen this example in the examples folder: https://codesandbox.io/s/tabs-760h9

**Reply by Biz-e-Services** [reply]
Yes, but as said I'm using tab's within tabs (so to speak" I think by using your suggestion "use v6 shouldUnregister" it is sort of working but I feel it's a bit clunky. I seem to get strange results when flicking between tab's (not forms) and saving that form (step 2).
On my onSubmit of Form 2 I have a check ...
`if (!safetyInvestigation) { toast.error('You have errors on the Safety Tab. You may have edited an entry. Please confirm your selections'); setError("safetyInvestigation")
         return null};`
which on second pass does not show any toaster but does disable the submit. So I will have to assume that the user would read the initial error toast msg and correct their entry. Any advise on a better way of doing it?

**Reply by bluebill1049** [reply]
I think tabs with tabs is the problem. I imagine your parent tab is actually for navigating the "page", I would use separate useForm for each "page" and the child tab is the real tab, which you can resolve by `shouldUnregister`.

**Reply by Biz-e-Services** [reply]
Seems to work. Thanks
